parser.js

```diff
@@ -841,16 +841,26 @@
     }
 
 
-    function parseDecimalEscape() {
+    function parseDecimalEscape(insideCharacterClass) {
       // DecimalEscape ::
       //      DecimalIntegerLiteral [lookahead ∉ DecimalDigit]
 
-      var res, match;
+      var res, match, from = pos;
+
+      if (hasUnicodeFlag) {
+        if (res = matchReg(/^\d/)) {
+          if (res[0] !== "0" || (res = matchReg(/^\d/)) ) {
+            bail("Invalid decimal escape in unicode mode", null, from, pos);
+          }
+          return createEscaped('null', 0x0000, '0', 1);
```

Comment:
Is `\000` also allowed in Unicode mode?

looks like it is treated as null further down

Reply:
I don't think it is allowed. Both Gecko and JSC throws on `/\000/u`.